host_core/lib/host_core.ex

```diff
@@ -181,9 +181,14 @@ defmodule HostCore do
   end
 
   defp write_json(config, file) do
-    case File.write(file, Jason.encode!(remove_extras(config))) do
-      {:error, reason} -> Logger.error("Failed to write configuration file #{reason}")
-      :ok -> Logger.info("Wrote #{inspect(file)}")
+    with :ok <- File.mkdir_p(Path.dirname(file)) do
+      case File.write(file, Jason.encode!(remove_extras(config))) do
```

Comment:
It's probably unnecessary, but have we considered adding `Jason.encode(_)` to the `with` statement to safely handle any encoding failures?

I can see this being more useful in regards to decoding unknown input, but it might help here too

Reply:
Most of the time I'd agree, but since we're the ones emitting the struct, and we know the output of `remove_extras` will always produce a JSON-encodable map, we're safe using the `!` here.